plugin/pkg/scheduler/scheduler.go

```diff
@@ -241,6 +242,39 @@ func (sched *Scheduler) assume(assumed *v1.Pod, host string) error {
 	// If the binding fails, scheduler will release resources allocated to assumed pod
 	// immediately.
 	assumed.Spec.NodeName = host
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeTopologyBinding) {
+		// AssumeVolume first
+		// If no volumes need binding, then continue with AssumePod.
+		// Otherwise, don't AssumePod, and continue to BindVolumes.
+		// After BindVolumes completes, the Pod will be sent back through the scheduler
+		// for a second pass where it can AssumePod once all the volumes are bound.
+		//
+		// TODO: can we distingush between fully bound and waiting for binding to complete?
+		// If it's waiting for binding to complete, then we can return early here instead of
+		// spawning a go routine for bindVolumes to do nothing...
+		bindingRequired, err := sched.config.VolumeBinder.AssumePodVolumes(assumed, host)
+		if err != nil {
+			sched.config.Error(assumed, err)
+			sched.config.Recorder.Eventf(assumed, v1.EventTypeWarning, "FailedScheduling", "AssumePod failed: %v", err)
+			sched.config.PodConditionUpdater.Update(assumed, &v1.PodCondition{
+				Type:    v1.PodScheduled,
+				Status:  v1.ConditionFalse,
+				Reason:  "SchedulerError",
+				Message: err.Error(),
+			})
+			return err
+		}
+		if bindingRequired {
+			if sched.config.Ecache != nil {
+				invalidPredicates := sets.NewString("NoVolumeBindConflict", "NoVolumeZoneConflict")
+				sched.config.Ecache.InvalidateCachedPredicateItemOfAllNodes(invalidPredicates)
+			}
+			return nil
```

Comment:
I think we should return an error and start a go routine to bind the volumes instead of doing the volume binding in the pod binding function.